PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/BasicLinkMySQLPanel.java

```diff
@@ -0,0 +1,277 @@
+import com.microsoft.azure.toolkit.intellij.link.LinkComposite;
+import com.microsoft.azure.toolkit.intellij.link.ModuleLinkConfig;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.form.AzureFormInput;
+import com.microsoft.azuretools.azurecommons.util.Utils;
+import com.microsoft.intellij.ui.components.AzureWizardStep;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.*;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+@Getter
+@Setter
+public class BasicLinkMySQLPanel<T extends LinkComposite<MySQLLinkConfig, ModuleLinkConfig>> extends JPanel implements AzureFormPanel<T> {
+    private SubscriptionComboBox subscriptionComboBox;
+    private ServerComboBox serverComboBox;
+    private DatabaseComboBox databaseComboBox;
+    private UsernameComboBox usernameComboBox;
+    private JPasswordField inputPasswordField;
+    private JPanel rootPanel;
+    private JTextField envPrefixTextField;
+    private PasswordSaveComboBox passwordSaveComboBox;
+    private JTextField urlTextField;
+    private JButton testConnectionButton;
+    private ModuleComboBox moduleComboBox;
+    private JTextPane testResultTextPane;
+    private JButton testResultButton;
+    private JLabel testResultLabel;
+    private JTextField portTextField;
+
+    private final Supplier<T> supplier;
+    private AzureWizardStep wizardStep;
+    private Project project;
+
+    private String urlHead = "jdbc:mysql://";
+    private String urlMiddle = ":3306/";
+    private String urlTail = "?serverTimezone=UTC&useSSL=true&requireSSL=false";
+
+    public BasicLinkMySQLPanel(Project project, final Supplier<T> supplier) {
+        super();
+        this.project = project;
+        this.supplier = supplier;
+        init();
+        initListeners();
+    }
+
+    private void init() {
+        Dimension lastColumnSize = new Dimension(106, 30);
+        passwordSaveComboBox.setPreferredSize(lastColumnSize);
+        passwordSaveComboBox.setMaximumSize(lastColumnSize);
+        passwordSaveComboBox.setSize(lastColumnSize);
+        envPrefixTextField.setPreferredSize(lastColumnSize);
+        envPrefixTextField.setMaximumSize(lastColumnSize);
+        envPrefixTextField.setSize(lastColumnSize);
+        T config = supplier.get();
+        Optional.ofNullable(config.getService().getSubscription())
+                .ifPresent((subscription -> {
+                    this.subscriptionComboBox.setValue(new AzureComboBox.ItemReference<>(subscription.subscriptionId(), Subscription::subscriptionId));
+                    subscriptionComboBox.setForceDisable(true);
+                    subscriptionComboBox.setEditable(false);
```

Comment:
May set editable(false) in `setForceDisable()`

Reply:
setForceDisable lies in super AzureComboBox, i think i should be better to let it pure and simple.